modules/AlphaWalletFoundation/AlphaWalletFoundation/EtherClient/Models/GetErc20Allowance.swift

```diff
@@ -30,6 +30,7 @@ class GetErc20Allowance {
                 } else {
                     return (false, amount - allowance)
                 }
-            }.eraseToAnyPublisher()
+            }.receive(on: RunLoop.main)
```

Comment:
Why is the explict `.receive(on: .main)` needed?

Reply:
by default it was on `.global`, updating the ui should be on `.main`

Reply:
But `callSmartContract(withServer:contract:functionName:abiString:parameters:shouldDelayIfCached:)` is on `main` right?